torch/csrc/distributed/rpc/ProcessGroupAgent.cpp

```diff
@@ -182,20 +202,11 @@ void ProcessGroupAgent::listenLoop() {
 
     Message message = deserialize(ss);
```

Comment:
Ideally we'd like to move deserialize to the thread pool as well to unblock the next receive while we're deserializing the previous message. this will allow to increase parallelization (it won't decrease latency of a single call though ...).

Reply:
agree! we can add the message type into preamble for shutdown. Will make the change.